### DISCUSSION TITLE
Countdown timer

### DISCUSSION THREAD

**Discussion by ceyerkes** [twilio-labs/paste#2186] [Q&A] [answered]
Another deadline is approaching for customers that haven't registered for A2P, which includes a significant increase in fees. We're looking into how to increase engagement with the nudges that we currently have in place (alert banners), in addition to introducing more aggressive nudges (bigger CTA on messaging homepage, modal at login, etc). The product partners I'm working with are very interested in exploring a countdown module, and I'm wondering if a component like this exists, or if there's something that makes sense within Paste that I can leverage. Any help would be greatly appreciated - I can also attend office hours next week. Thanks!

**Comment by gloriliale** [comment]
Hi @ceyerkes! I've asked our team about this and will get back to you asap with our thoughts!

**Reply by ceyerkes** [reply]
Thank you!

**Comment by gloriliale** [comment]
Hi @ceyerkes! Is there any chance you could post, or send me directly if it's sensitive, links to the current designs you're trying to increase engagement on? It'd be really helpful!

**Reply by ceyerkes** [reply]
Yes of course! Here's the [link](https://www.figma.com/file/HBX2wnEszgWwf6iuesSrAY/?node-id=16%3A91603). Happy to provide more context, just let me know. Thanks!

**Reply by gloriliale** [reply]
This is great, thank you! The main thing that stands out from our perspective is that it looks like the countdown of "days left to register" is inside of a Toast right now, but we'd suggest using a non-dismissable alert there. If you'd like to chat more and are available during tomorrow's Office Hours, we do have one more slot open!

**Reply by ceyerkes** [reply]
We're on a very tight timeline right now, so I'm trying to figure out how to use existing components to convey the type of urgency we need, hence the countdown + toast designs idea. We use non-dismissable alerts that communicate the deadline, but unfortunately they haven't been very effective. I can't make office hours tomorrow, but would love to in the future - if anything comes to mind that you think might work for this use case, please let me know!

**Reply by gloriliale** [reply]
Got it. It might be worth seeing how technically feasible it'd be to have the non-dismissable alert follow a user throughout console, simply because toast is more for feedback on something a user's immediately done (more info in our sections about [toasts vs. alerts vs. inline messaging](https://paste.twilio.design/components/toast/#toast-vs-alerts-vs-inline-messaging) and [notifications and feedback pattern](https://paste.twilio.design/patterns/notifications-and-feedback) ). I'll be sure to post if we think of any other mechanism though!

**Reply by ceyerkes** [reply]
The current behavior of the alert is that it IS dismissible in console. There's concern around having an alert that follows a user around console and isn't dismissible. Do we have any other components that you can think of that would be more in-line with prompting a user to take action? Repurposing the toast is the closest thing we've come up with so far, but I totally understand the concern that it's not the perfect solution, and toasts aren't meant for that type of feedback.

**Reply by ceyerkes** [reply]
One other question: do we have anything like this global banner that sits above the top nav? Or could we repurpose our current alert into something like this?
![Screen Shot 2022-02-08 at 2 47 38 PM](https://user-images.githubusercontent.com/87141551/153081655-b8a0c433-7b36-4c70-a7c8-acaa104d3cf0.png)

**Reply by gloriliale** [reply]
From our side, that would work, and a non-dismissable Alert would be the best component to use for it. The technical feasibility of putting it above console's masthead would be the only thing blocking it. I think @hharnisc and co. would be the ones w/ more details on that side. From a design perspective though, this would be our direction.

**Reply by ceyerkes** [reply]
Thank you @gloriliale! Really appreciate your help.

**Comment by hharnisc** [comment, marked as answer]
To close the loop here from the Console side. Based on the UX guidelines in Paste we went with the Notification and we have most of the moving parts in place to trigger the Notification to show and hide. We've got an open PR and collecting feedback from @ceyerkes and team 🙌
